core/src/Cabana_VerletList.hpp

```diff
@@ -375,6 +329,91 @@ struct VerletListBuilder
             } );
     }
 
+    // Neighbor count team vector operator (only used for CSR lists).
+    KOKKOS_INLINE_FUNCTION void
+    neighbor_reduce( const typename CountNeighborsPolicy::member_type &team,
+                     const std::size_t pid, const int i, const int j,
+                     const int k, int &cell_count, TeamVectorOpTag ) const
+    {
+        // Cache the particle coordinates (again).
+        double x_p = position( pid, 0 );
+        double y_p = position( pid, 1 );
+        double z_p = position( pid, 2 );
+
+        std::size_t a_offset = linked_cell_list.binOffset( i, j, k );
+
+        Kokkos::parallel_reduce(
+            Kokkos::ThreadVectorRange( team,
```

Comment:
I think you can have another reduce abstraction here because this inner. A lot of code is still repeated between these two functions. Perhaps at least abstract the loop body to take in a `local_count` by reference - you can then either just pass in which ever value it reduces to.

Reply:
Took care of this in a similar way